ui - Alert and platform libraries + rebranding

JBWatenbergScality opened the pull request:
**Component**:

Shell-ui

**Context**: 

Solutions UI may need to fetch alerts. Sell-ui Alerts library goal is to expose alerts hooks to simplify fetching alerts.

**Acceptance criteria**: 

Closes: #3259

JBWatenbergScality committed:
shell-ui: Move bundles to sub folders

JBWatenbergScality committed:
shell-ui: Swith whatwg-fetch to node-fetch to polyfill fetch in node (refs: https://github.com/mswjs/msw/issues/448)

JBWatenbergScality committed:
shell-ui: Add env to start dev preview of alerts or navbar

JBWatenbergScality committed:
shell-ui: Add alert library

bert-e commented:
# Hello jbwatenbergscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality

ChengYanJin commented on shell-ui/src/alerts/services/alertUtils.js:
```diff
@@ -0,0 +1,242 @@
+  } else return false;
+};
+
+export const getHealthStatus = (
```
Not sure if all the functions in the alertUtils are needed by `shell-ui`. IMO this `getHealthStatus` is not needed here.

ChengYanJin commented on shell-ui/package.json:
```diff
@@ -4,7 +4,8 @@
   "scripts": {
     "test": "jest",
-    "start": "webpack serve --config webpack.config.dev.js",
+    "start:navbar": "webpack serve --config webpack.config.dev.js --env=entry=navbar",
+    "start:alerts": "webpack serve --config webpack.config.dev.js --env=entry=alerts",
```
I am wondering do we want to run the navbar and alerts all together?

JBWatenbergScality committed:
shell-ui: fixup alert library

JBWatenbergScality committed:
ui: Use shell-ui alert library

ChengYanJin commented on ui/src/containers/AlertProvider.js:
```diff
@@ -1,45 +1,98 @@
 };
+
+function ErrorFallback({ error, resetErrorBoundary }) {
+  //Todo redirect to a beautiful error page
```
The error pages in `core-ui` is ready ^^ (We need to release a new version in core-ui though)

ChengYanJin commented on shell-ui/src/alerts/Alerts.spec.js:
```diff
@@ -0,0 +1,134 @@
+    await waitForNextUpdate();
+
+    //V
+    expect(result.current.alerts.map(alert => alert.originalAlert)).toStrictEqual(alerts);
```
not sure the test will pass in CI because of the `endsAt`. I'll let you check it^^.

ChengYanJin committed:
shell-ui: Add alerthook and expose in library

Refs: #3257

ChengYanJin committed:
shell-ui: Add k8s service

Check if the k8s call carry the `id_token`, otherwise listen to the
navbarElement AUTHENTICATED_EVENT

Refs: #3275

JBWatenbergScality committed:
shell-ui: Add a getIdToken method on the navbar element to allow synchronization of elements mounted after the navbar

JBWatenbergScality committed:
shell-ui: Rebranding base modifications

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/shell-ui-alert-provider` and the
destination branch `development/2.9`.

Please resolve the conflict on **the feature branch** (`improvement/shell-ui-alert-provider`).

```bash
 $ git fetch
 $ git checkout origin/improvement/shell-ui-alert-provider
 $ git merge origin/development/2.9
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/shell-ui-alert-provider
```

ChengYanJin committed:
shell-ui: Add test for getNodesCount

Refs: #3257

ChengYanJin committed:
shell-ui: Add enabled option in query object

To disable the query if token is not provided

Refs: #3257

JBWatenbergScality committed:
sheel-ui: Fix shell-ui platform library

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality

ChengYanJin commented on shell-ui/package.json:
```diff
@@ -27,15 +29,16 @@
   },
   "dependencies": {
-    "@scality/core-ui": "github:scality/core-ui.git#v0.13.0",
+    "@scality/core-ui": "github:scality/core-ui.git#improvement/doc-color",
```
FYI:
I just released a new version in `core-ui` v0.15.0 

(The PR looks good to me!)
